src/video/wayland/SDL_waylandvideo.c

```diff
@@ -369,7 +370,54 @@ static void Wayland_DeleteDevice(SDL_VideoDevice *device)
     SDL_WAYLAND_UnloadSymbols();
 }
 
-static SDL_VideoDevice *Wayland_CreateDevice(void)
+typedef struct
+{
+    SDL_bool has_fifo_v1;
+    SDL_bool has_commit_timing_v1;
+} SDL_WaylandPreferredData;
+
+static void wayland_preferred_check_handle_global(void *data, struct wl_registry *registry, uint32_t id,
+                                                  const char *interface, uint32_t version)
+{
+    SDL_WaylandPreferredData *d = data;
+
+    if (SDL_strcmp(interface, "wp_fifo_manager_v1") == 0) {
+        d->has_fifo_v1 = SDL_TRUE;
+    } else if (SDL_strcmp(interface, "wp_commit_timing_manager_v1") == 0) {
```

Comment:
I suppose the `else` here is unintentional? Otherwise the later `preferred_data.has_fifo_v1 && preferred_data.has_commit_timing_v1` will never evaluate to true.

Reply:
Never mind, I should not read code close to bed time :)